### PR TITLE
[shelly] TRV+Motion2 support, various fixes & enhancements

### PR DESCRIPTION
**Pull request by markus7017**
Includes
- Shelly TRV support incl. new channels
- Shelly Motion 2 support (like Motion 1 + temp channel)
- Translations updated
- Meta data added to XML (category etc.)
- README updated
- Several bugfixes

### CODE REVIEW

**Comment by hmerk** [comment]
@markus7017 As I learned, milestones are added by maintainers after merge, therefore I removed it.

**Comment by mhupfauer** [comment]
Is there an ETA when this will be merged?

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:163]: Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug. See https://next.openhab.org/docs/developer/guidelines.html#f-logging

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:359]: See above. Logging to Info. Please check all.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:537]: Can the single semicolon be removed?

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyComponents.java
Comment (fwolter) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyComponents.java:330]: Can the comments be removed?

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties
Comment (fwolter) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: @openhab/add-ons-maintainers There are quite a lot changes in the translation in this PR. If I understood correctly, translations should only be handled in Crowdin, otherwise they will be overwritten in the repo. How do we handle this?

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: Most of the translations are even no changed. I moved all str resources from xml to the .properties file. Changes must not be overwritten depending on tooling. Please advise how to proceed

Comment (lolodomo) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: First, the Shelly binding is one of the bindings for which i18n is not yet properly implemented.
What is done here is surprising. I am not a fan of replacing all labels/descriptions in XML by `@text` entries but this is a way to do. What is strange here is that it looks like that `@text` entries look like the default paths that will have been considered even without the `@text`. So it was relatively useless to update all the XML.
**I would suggest to undo these changes from this PR and do it in a separate PR.** This can be done automatically by simply run a maven command. Using this command has the big advantage as it can be rerun later to simply add a new entry.
And regarding the German translations, it is better to update them in Crowdin in a second time. Because if the PR is merged and no manual action is done in Crowdin, there is a risk to loose them in the future.  An exception could be done as this is more or less a first massive injection of translations.

My suggestion: separate PR and simply use the maven command to generate the default translations.

**Comment by mschoenbein** [comment]
Is there an ETA when this will be merged? Wait for TVR support.

**Comment by hmerk** [comment]
> Is there an ETA when this will be merged? Wait for TVR support.

No, in general, we don't have ETAs, as all maintainers and contributors are doing this on voluntary basis in their spare time.
If you want to use the feature right now follow the Shelly binding topic in our community and install the latest dev version, like I do. TRV support is woking great.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:163]: Many people don't recognize the Shelly Manager UI, because OH doesn't provide an interface for UI extensions provided by the binding (as already requested a long time ago) and many don't reads the manual (RTFM). Therefore I'm adding this info to the log to promote this useful feature.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:359]: Doesn't it make use to make the user aware of a mis-configuration. Otherwise nothing happens and there is no indication in the log with default config. How to find this out without a log message?

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:537]: fixed

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyComponents.java
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyComponents.java:330]: done

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: - " is not yet properly implemented." - where is a proper way described. 
- "This can be done automatically by simply run a maven command. " - which one?
- "And regarding the German translations, it is better to update them in Crowdin in a second time. " - looking to the volume of translations this is a huge effort. Where is the Crowdin integration described? Where documented that this is a formal requirement?
- "Yes please"
- "My suggestion: separate PR" - again, a huge effort. Everything has to be done manually. I think we should work on another one to make the move to CrowedIn.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (fwolter) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:356]: If this is supposed to warn the user something is wrong, this should be WARN.

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties
Comment (ijasan) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: I supported @markus7017 on the translations, I could also support on moving resources to CrowdIn, but I also recommend the exception for this PR to avoid blocking the process. If the @text statements are not nessesary we could remove them, but maybe @fwolter could also proceed.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:356]: ok, I reviewed and adjusted all log levels in this module

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: @lolodomo any advice?

Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: @fwolter @lolodomo who do we solve this issue

**Comment by markus7017** [comment]
@fwolter I'm ready to continue

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties
Comment (fwolter) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly_de.properties:1]: I'd suggest to populate Crowding correctly once this is merged. Also because @ijasan offered his support.

**Review by fwolter — APPROVED** [review]

**Pull request merged**
